### PR TITLE
Adding column names to the log entry.

### PR DESCRIPTION
**Pull request by irieksts**

### CODE REVIEW

File: src/main/java/com/hmsonline/cassandra/triggers/LogEntryStore.java [hunk 1]
```diff
@@ -78,7 +78,11 @@ public static String getHostName() throws SocketException {
                     }
                 }
             }
-            hostName = hostName.substring(0, hostName.indexOf('.'));
+            if(hostName.indexOf('.') > -1) {
+            	hostName = hostName.substring(0, hostName.indexOf('.'));
+            } else if(hostName.indexOf(':') > -1) {
```
Comment (boneill42) [inline src/main/java/com/hmsonline/cassandra/triggers/LogEntryStore.java:83]: I don't think you want to do a substring here.  If you do that, you'll only get the first bytes of an IPV6 address, which may collide with other hosts.  Instead, just replace ":" with empty (or something else).

**Comment by ducquoc** [comment]
The getHostName() using NetworkInterface can retrieve correct hostname on physical machines but on VM sometimes it resolves to IPv4 or IPv6. If it resolve to 192.168.22.208 then the triggers can have lots of logs:

```
ERROR [Thread-42] 2012-04-05 22:38:41,891 TriggerTask.java (line 56) Could not execute triggers.
InvalidRequestException(why:unconfigured columnfamily CommitLog_192)
        at org.apache.cassandra.thrift.ThriftValidation.validateColumnFamily(ThriftValidation.java:129)
        at org.apache.cassandra.thrift.CassandraServer.get_range_slices(CassandraServer.java:672)
        at com.hmsonline.cassandra.triggers.CommitLog.getPending(CommitLog.java:80)
        at com.hmsonline.cassandra.triggers.TriggerTask.run(TriggerTask.java:38)
        at java.lang.Thread.run(Thread.java:662)
ERROR [Thread-42] 2012-04-05 22:38:41,891 TriggerTask.java (line 56) Could not execute triggers.
InvalidRequestException(why:unconfigured columnfamily CommitLog_192)
        at org.apache.cassandra.thrift.ThriftValidation.validateColumnFamily(ThriftValidation.java:129)
        at org.apache.cassandra.thrift.CassandraServer.get_range_slices(CassandraServer.java:672)
        at com.hmsonline.cassandra.triggers.CommitLog.getPending(CommitLog.java:80)
        at com.hmsonline.cassandra.triggers.TriggerTask.run(TriggerTask.java:38)
        at java.lang.Thread.run(Thread.java:662)
```

I also tried java.net.InetAddress.getLocalHost().getHostName() , and it seems to always get the hostname (on VM) , but fully-qualified domain name on physical machines which is not so good as well.

**Comment by boneill42** [comment]
Thanks Duc.

Your right.  We should change it to use the whole IP address.
We can replace "." with "_".

I'll make the change.

-brian

## 

Brian O'Neill
Lead Architect, Software Development
Health Market Science | 2700 Horizon Drive | King of Prussia, PA 19406
p: 215.588.6024
www.healthmarketscience.com

On 4/12/12 12:37 PM, "Duc M. Quoc"
<reply+i-3931864-0f403dfde1f53fa0e2f362dd9d7e2bbacf5baf5e-897229@reply.gith
ub.com> wrote:

**Comment by boneill42** [comment]
Fixed in:
https://github.com/hmsonline/cassandra-triggers/pull/32

-brian

## 

Brian O'Neill
Lead Architect, Software Development
Health Market Science | 2700 Horizon Drive | King of Prussia, PA 19406
p: 215.588.6024
www.healthmarketscience.com

**Comment by ducquoc** [comment]
thanks Brian

**Pull request merged**